core/solver/multigrid.cpp

```diff
@@ -563,14 +563,39 @@ void Multigrid::generate()
             auto exec = this->get_executor();
             // default coarse grid solver, direct LU
             // TODO: maybe remove fixed index type
-            auto gen_default_solver = [&] {
-                return experimental::solver::Direct<value_type, int32>::build()
-                    .with_factorization(
-                        experimental::factorization::Lu<value_type,
+            auto gen_default_solver = [&]() -> std::unique_ptr<LinOp> {
+                // TODO: unify when dpcpp supports direct solver
+                if (dynamic_cast<const DpcppExecutor*>(exec.get())) {
+                    using absolute_value_type = remove_complex<value_type>;
+                    return solver::Gmres<value_type>::build()
+                        .with_criteria(
+                            stop::Iteration::build()
+                                .with_max_iters(matrix->get_size()[0])
```

Comment:
Did you test the default behavior on a decently-sized matrix? If we have the choice between taking forever and producing a precise result, and returning an imprecise result after a shorter time, I would prefer going for the latter.

Reply:
Not sure what I did here, but I wanted it to be `min(100, matrix->get_size()[0])`. I just wanted to make sure that for small matrices, we don't use more krylov vectors than the matrix size.

Reply:
I think as default, we should solve exactly on the coarsest level. If that can't be achieved in reasonable time, then the problem might be too difficult for the default settings and the user needs to specify the solver anyway.